lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -198,23 +208,36 @@ def mock_run(args, cwd, stderr):
         tree.create("ABC", source_tarball)
 
         tarball = tree.find_dataset(md5)
+        assert tarball.resource_id == md5
+        assert tarball.resource_id in tree
         tree.unpack(md5)
+        assert tarball.unpacked
 
     def test_unpack_with_exception(
         self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
     ):
         """Test to check the unpacking functionality of the FileTree when
         an Exception occured"""
         source_tarball, source_md5, md5 = tarball
-
-        def mock_run(args, cwd, stderr):
+        tarball_name = source_tarball.name[0 : -len(".tar.xz")]
+
+        def mock_run(args, cwd, stdin, stderr, stdout):
+            assert args == [
+                "tar",
+                "-xf",
+                (server_config.ARCHIVE / "ABC" / source_tarball.name),
+            ]
+            assert cwd == server_config.INCOMING / "ABC" / tarball_name
+            assert stderr == subprocess.DEVNULL
             raise subprocess.TimeoutExpired("tar", 43)
 
         monkeypatch.setattr("subprocess.run", mock_run)
         tree = FileTree(server_config, make_logger)
         tree.create("ABC", source_tarball)
 
         tarball = tree.find_dataset(md5)
+        assert tarball.resource_id == md5
+        assert tarball.resource_id in tree
         with pytest.raises(TarballUnpackError) as exc:
             tree.unpack(md5)
         assert (
```

Comment:
Analogous to my previous comment, we don't really need the assertions at lines 239-240 (although they are OK), because they don't test anything different from the check at lines 211-212.  The key part of this test scenario is the assertion at line 243 (which is triggered by the fact that the mock'd `subprocess.run()` produces a different result); otherwise, this test is pretty much the same as the previous one, so we don't need to duplicate everything.  (Ditto lines 273-274.)